src/v/iceberg/rest_catalog.cc

```diff
@@ -50,6 +51,10 @@ struct http_error_mapping_visitor {
     errc operator()(const http::url_build_error&) const {
         return unexpected_state;
     }
+
+    errc operator()(const rest_client::aborted_error&) const {
+        return shutting_down;
```

Comment:
not related to this PR, but `using enum` style looks very sharp for an error code in particular.

Reply:
😄 yea I'm pretty fond of it these days. I will say I prefer _not_ using it when there are multiple enum types in close proximity -- that gets confusing real fast

Reply:
yeah exactly. specifically much nicer than typing (and reading) `my_very_descriptive_namespace::errc::my_very_descriptive_id` over and over again 🙂